File: projects/blognami/lib/blognami/models/page.rb

require "blognami/models/pageable"

Slick::Database::Row.register "page" do
  
    include :pageable

    belongs_to :user

    must_not_be_blank :user_id
    must_not_be_blank :title

    before_validation do
        self.published_at = Time.now if published && published_at.nil?
    end

end

File: packages/blognami/lib/models/page.js

export default {
    meta(){
        this.include('pageable');
        
        this.belongsTo('user');

        this.mustNotBeBlank('userId');
        this.mustNotBeBlank('title');

        this.beforeValidation(function(){
            if(this.published && !this.publishedAt){
                this.publishedAt = new Date();
            }
        })
    }
};
